packages/itwin/grouping-mapping-widget/src/components/Mappings/hooks/useFetchExtractionStatus.ts

```diff
@@ -0,0 +1,64 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import type { IExtractionClient } from "@itwin/insights-client";
+import { useQuery } from "@tanstack/react-query";
+import type { GetAccessTokenFn } from "../../context/GroupingApiConfigContext";
+import type { ExtractionMessageData } from "../Extraction/ExtractionMessageModal";
+import type { ExtractionStatusData } from "../Extraction/ExtractionStatusIcon";
+
+export const useFetchExtractionStatus = ({
+  iModelId,
+  getAccessToken,
+  extractionClient,
+}: {
+  iModelId: string;
+  getAccessToken: GetAccessTokenFn;
+  extractionClient: IExtractionClient;
+}) => {
+  return useQuery({
+    queryKey: ["extractionStatus", iModelId],
```

Comment:
The `extractionStatus` can conflict with a specific extraction, so I would recommend using a more specific name. Since this is tracking the latest iModel extraction, I would go with `iModelExtractionStatus`. Then if you would be tracking a specific extraction (not by iModelId, but by extractionId) you could use `extractionStatus` with the extractionId.

Reply:
Changed to `iModelExtractionStatus`